pkg/processor/k8sprocessor/kube/client.go

```diff
@@ -247,21 +247,42 @@ func (c *WatchClient) deleteLoop(interval time.Duration, gracePeriod time.Durati
 	}
 }
 
-// GetPod takes an IP address or Pod UID and returns the pod the identifier is associated with.
-func (c *WatchClient) GetPod(identifier PodIdentifier) (*Pod, bool) {
+// getPod takes an IP address or Pod UID and returns the pod the identifier is associated with.
+func (c *WatchClient) getPod(identifier PodIdentifier) (*Pod, bool) {
 	c.m.RLock()
 	defer c.m.RUnlock()
 	pod, ok := c.Pods[identifier]
-	if ok {
-		if pod.Ignore {
-			return nil, false
-		}
+	if !ok {
+		observability.RecordIPLookupMiss()
+		return nil, ok
+	}
+	if pod.Ignore {
+		return nil, false
+	}
+	return pod, true
+}
 
-		c.updatePodOwnerMetadata(pod)
-		return pod, ok
+// GetPodAttributes takes an IP address or Pod UID and returns the metadata attributes of the Pod the
+// identifier is associated with
+func (c *WatchClient) GetPodAttributes(identifier PodIdentifier) (map[string]string, bool) {
+	pod, ok := c.getPod(identifier)
+	if !ok {
+		return nil, false
 	}
-	observability.RecordIPLookupMiss()
-	return nil, false
+	ownerAttributes := c.getPodOwnerMetadataAttributes(pod)
+
+	// we need to take a lock here because pod.Attributes may be modified concurrently
+	// TODO: clean up the locking in these functions
```

Comment:
which functions?

Reply:
This one and the ones it calls, made that clearer in the comment.